### PR TITLE
[Collections] static analyzer fix and cleanup.

### PR DESCRIPTION
**Pull request by ianegordon**
Work in Progress.

### CODE REVIEW

**Comment by ianegordon** [comment]
This is my initial pass on fixing the static analyzer and some general cleanup.

Please let me know if anything looks suspicious or non-standard.

File: components/Collections/src/MDCCollectionViewController.m
Comment (willlarche) [inline components/Collections/src/MDCCollectionViewController.m:350]: alignment

File: components/Collections/src/MDCCollectionViewController.m
Comment (willlarche) [inline components/Collections/src/MDCCollectionViewController.m]: Is this important?

**Review by willlarche — CHANGES_REQUESTED** [review]
Run format_all over this for a couple alignment things.

There's new nslogs. Were they needed?

File: components/Collections/src/MDCCollectionViewController.m
Comment (ianegordon) [reply components/Collections/src/MDCCollectionViewController.m:350]: All NSLogs (or any other lines with KM / KillMe) will be removed before merge.

Comment (ianegordon) [reply components/Collections/src/MDCCollectionViewController.m:350]: (The NSLogs will be useful for testing, so I'll keep them until everyone is satisfied)

Comment (ianegordon) [reply components/Collections/src/MDCCollectionViewController.m:350]: format_all run.

Comment (willlarche) [reply components/Collections/src/MDCCollectionViewController.m:350]: 👌

**Review by willlarche — APPROVED** [review]

**Comment by willlarche** [comment]
Still needs @chriscox ' approval.

File: components/Collections/examples/CollectionsEditingExample.m
Comment (ianegordon) [inline components/Collections/examples/CollectionsEditingExample.m:115]: I am not sure of the best way to handle this.
An example should not have to deal with custom MDCCollection views.

Should we add a way to query your super class for a reuseIdentifier based on kind?
- (NSString *)reuseIdentifierForKind:(NSString *)kind

We cannot just invoke a super implementation as this method must return a nonnull value.

Comment (ShepJGoogle) [reply components/Collections/examples/CollectionsEditingExample.m:115]: This will alter the use of the collection. In the MDCCollectionViewController class, within this method we also call updateControllerWithInfoBar:(MDCCollectionInfoBar *)infoBar which in short, assigns the supp view as a property to the controller so it can update it with each selection/deselection. So it's not just about updating the identifiers.

Comment (ShepJGoogle) [reply components/Collections/examples/CollectionsEditingExample.m:115]: There's also some delegate methods for the MDCCollectionInfoBarViewDelegate protocol assigned to the info bars which the controller needs to be able to handle.

**Review by ShepJGoogle — CHANGES_REQUESTED** [review]

**Comment by ShepJGoogle** [comment]
I think something to consider is decoupling our editing info bars from the collection view data source implementation and making them standalone views as part of the MDCCollectionViewController class. At a high level, the supplementary views we currently register as info header and info footer bars can be made as simple uiviews which we would insert as regular subviews in the view. It would remove all of the overheard and trickiness around having to inject the logic into the supplementary view data source method.

**Comment by ianegordon** [comment]
Justin's comment about decoupling the info bars makes a lot of sense.
Are there any downsides?
@chriscox  ?

**Comment by chriscox** [comment]
@ShepJGoogle @ianegordon I do like Justin's idea about decoupling the inforbars.  Some of the requirements might be:

- What about creating a separate infobar view controller. One that handles both header/footer placement and visibility delegates. This controller is responsible for show/hide animations.
- Either `MDCCollectionView` or clients subclass would be required to trigger the infobar controller when entering edit more. I think if we handle in `MDCCollectionView`, then can be customized or overridden by client?
- The infobar background color, text color, and text could all be made customizable.
- Will need to provide delegates to forward touch events back to collection view controller.

**Comment by ianegordon** [comment]
I will open an issue to investigate changing our infobar handling.

I've created #1316 as a simplified PR to fix the analyzer warning.

**Comment by ianegordon** [comment]
Infobar issue #1317

**Pull request closed without merging**
